internal/graph/check_fast_path.go

```diff
@@ -616,6 +616,47 @@ func (c *LocalChecker) resolveFastPath(ctx context.Context, leftChans []chan *it
 	return res, ctx.Err()
 }
 
+func (c *LocalChecker) checkUsersetFastPathV2(ctx context.Context, req *ResolveCheckRequest, iter storage.TupleKeyIterator) (*ResolveCheckResponse, error) {
+	ctx, span := tracer.Start(ctx, "checkUsersetFastPathV2")
+	defer span.End()
+
+	typesys, _ := typesystem.TypesystemFromContext(ctx)
+	objectType := tuple.GetType(req.GetTupleKey().GetObject())
+
+	cancellableCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	directlyRelatedUsersetTypes, _ := typesys.DirectlyRelatedUsersets(objectType, req.GetTupleKey().GetRelation())
+
+	leftChans := make([]chan *iteratorMsg, 0)
```

Comment:
nit: 

```suggestion
	leftChans := make([]chan *iteratorMsg, 0, len(directlyRelatedUsersetTypes))
```

Reply:
Can't. Had this in the original version of TTU but caused goroutine leak given that the consumer gets the events out of order. So the close can be received before processing all messages.